pkg/controllers/provisioning/scheduling/provisioner_instance_selection_test.go

```diff
@@ -21,6 +21,7 @@ import (
 	"github.com/mitchellh/hashstructure/v2"
```

Comment:
Have we covered all of the test cases here with instance selection and the provisioner test suites? I don't see any new tests beyond in the `resources/suite_test.go` file that ensure that we are doing the correct calculation for pod resource requests while scheduling. I'd expect to see something like a few It() blocks that enforce some of the test cases that I called out in my initial review: https://github.com/aws/karpenter-core/pull/569#discussion_r1357770428

Reply:
I have configured "```should schedule on an instance with enough resources```" at ```provisioner_instance_selection.go``` and ```provisioner_instance_selection.go``` its sort of confirmation for restartPolicy consideration . maybe we could extend there other than that place what else you suggest ?

Reply:
I think that we should probably add tests to `provisioner_test.go` that specifically test these scenarios. Mentioned in the comment that I linked (https://github.com/aws/karpenter-core/pull/569#discussion_r1357770428) which should give you some guidance around some test cases that would be good to add to `provisioner_test.go`. Probably don't need all of those tests, but some subset of those tests should give us greater confidence that our logic is correct.